manager/state/raft/raft.go

```diff
@@ -340,7 +353,23 @@ func (n *Node) done() {
 // cluster state. If no saved state exists, it starts a single-node cluster.
 func (n *Node) Run(ctx context.Context) error {
 	ctx = log.WithLogger(ctx, logrus.WithField("raft_id", fmt.Sprintf("%x", n.Config.ID)))
-	defer n.done()
+	ctx, cancel := n.WithContext(ctx)
```

Comment:
But the only place we close `n.stopped` is inside `n.stop()`, which is called from the `defer`. So `WithContext` seems completely redundant to me.